e2e/portalicious/pages/RegistrationsPage.ts

```diff
@@ -0,0 +1,64 @@
+import { expect, Locator } from '@playwright/test';
+import { Page } from 'playwright';
+
+import BasePage from './BasePage';
+
+class RegistrationsPage extends BasePage {
+  readonly page: Page;
+  readonly tableRows: Locator;
+  readonly selectAllPasCheckbox: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+    this.tableRows = this.page.locator('table tbody tr');
+    this.selectAllPasCheckbox = this.page.getByRole('cell', {
+      name: 'All items unselected',
+    });
+  }
+
+  async selectAllPAs() {
+    await this.selectAllPasCheckbox.click();
+  }
+
+  async selectBulkAction(action: string) {
+    await this.page.getByRole('button', { name: action }).click();
+  }
+
+  async selectCustomMessage() {
+    await this.page
+      .locator('label')
+      .filter({ hasText: 'Custom message' })
+      .click();
+  }
+
+  async typeMessage(message: string) {
+    await this.page.fill('textarea', message);
+  }
+
+  async selectTemplatedMessage(messageTemplate: string) {
+    await this.page.getByLabel('Choose message').click();
+    await this.page.getByLabel(messageTemplate, { exact: true }).click();
+  }
+
+  async clickContinueToPreview() {
+    await this.page
+      .getByRole('button', { name: 'Continue to preview' })
+      .click();
+  }
+
+  async validateMessagePresent(message: string) {
+    await this.page.waitForTimeout(500);
+    const textboxValue = await this.page.$eval(
+      'textarea',
+      (textarea) => textarea.value,
+    );
+    expect(textboxValue).toContain(message);
+  }
```

Comment:
I'm afraid that this function could lead to a false positive, because we wouldn't detect if we are showing the right component.

Could we also check that the textarea is disabled? Maybe that would be sufficient.

Reply:
In this case there is only one component that checks for the specific templated or typed text. Saying that the default behaviour of the text area is to be "disabled" so no edits can be done during the preview phase. I added additional check so we can secure that text area behaves as supposed.